flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -921,11 +921,20 @@ public ActorGateway createActivatorDeactivator(ActorSystem actorSystem, UUID lea
 	// ------------------------------------------------------------------------
 
 	private class ScheduledTrigger extends TimerTask {
+		private final long processingInterval;
+		private long lastCheckpoint = 0;
+
+		public ScheduledTrigger(long guaranteedInterval) {
+			this.processingInterval = guaranteedInterval;
+		}
 
 		@Override
 		public void run() {
 			try {
-				triggerCheckpoint(System.currentTimeMillis());
+				long time = lastCheckpoint;
```

Comment:
where do you update `lastCheckpoint`? If I'm not mistaken, then it should always stay `0`.